tests/Avalonia.Controls.UnitTests/VirtualizingStackPanelTests.Buffered.cs

```diff
@@ -0,0 +1,2321 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.Diagnostics;
+using System.Linq;
+using Avalonia.Collections;
+using Avalonia.Controls.Presenters;
+using Avalonia.Controls.Primitives;
+using Avalonia.Controls.Templates;
+using Avalonia.Data;
+using Avalonia.Input;
+using Avalonia.Layout;
+using Avalonia.Media;
+using Avalonia.Styling;
+using Avalonia.UnitTests;
+using Avalonia.VisualTree;
+using Xunit;
+
+#nullable enable
+
+namespace Avalonia.Controls.UnitTests
+{
+    public class VirtualizingStackPanelWithBufferTests : ScopedTestBase
+    {
+        private static FuncDataTemplate<ItemWithHeight> CanvasWithHeightTemplate = new((_, _) =>
+            new Canvas
+            {
+                Width = 100,
+                [!Layoutable.HeightProperty] = new Binding("Height"),
+            });
+
+        private static FuncDataTemplate<ItemWithWidth> CanvasWithWidthTemplate = new((_, _) =>
+            new Canvas
+            {
+                Height = 100,
+                [!Layoutable.WidthProperty] = new Binding("Width"),
+            });
+
+        
+        [Fact]
+        public void When_Vertical_Calculates_ViewPort_At_Start_Of_List()
+        {
+            // Arrange
+            using var app = App();
+            
+            var items = Enumerable.Range(0, 100).Select(x => new ItemWithHeight(x)).ToList();
+            
+            // Act
+            var (target, scroll, itemsControl) = CreateTarget<ItemsControl,VirtualizingStackPanelCountingMeasureArrange>(items: items, itemTemplate: CanvasWithHeightTemplate);
```

Comment:
Nit: Some lines are extremely long (>210 chars): please follow the [style guidelines](https://github.com/AvaloniaUI/Avalonia/blob/master/CONTRIBUTING.md#style) (we don't have many). I have a 4k monitor and I need to scroll to see the end of these lines ;)

Reply:
I introduced line breaks